src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs

```diff
@@ -3015,6 +3015,506 @@ void M()
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(23394, "https://github.com/dotnet/roslyn/issues/23394")]
+        public async Task QueryMethodinfoJoin3()
+        {
+            await TestInMethodAsync(
+@"
+            var q = from i1 in new int[0]
+                    join i2 in new int[0] $$on i1 equals i2
+                    select i1;
+",
+            MainDescription($"({CSharpFeaturesResources.extension}) IEnumerable<int> IEnumerable<int>.Join<int, int, int, int>(IEnumerable<int> inner, Func<int, int> outerKeySelector, Func<int, int> innerKeySelector, Func<int, int, int> resultSelector)"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(23394, "https://github.com/dotnet/roslyn/issues/23394")]
+        public async Task QueryMethodinfoJoin4()
+        {
+            await TestInMethodAsync(
+@"
+            var q = from i1 in new int[0]
+                    join i2 in new int[0] on i1 $$equals i2
+                    select i1;
+",
+            MainDescription($"({CSharpFeaturesResources.extension}) IEnumerable<int> IEnumerable<int>.Join<int, int, int, int>(IEnumerable<int> inner, Func<int, int> outerKeySelector, Func<int, int> innerKeySelector, Func<int, int, int> resultSelector)"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(23394, "https://github.com/dotnet/roslyn/issues/23394")]
+        public async Task QueryMethodinfoJoinInto1()
+        {
+            await TestInMethodAsync(
+@"
+            var q = from i1 in new int[0]
+                    $$join i2 in new int[0] on i1 equals i2 into g
+                    select g;
+",
+            MainDescription($"({CSharpFeaturesResources.extension}) IEnumerable<IEnumerable<int>> IEnumerable<int>.GroupJoin<int, int, int, IEnumerable<int>>(IEnumerable<int> inner, Func<int, int> outerKeySelector, Func<int, int> innerKeySelector, Func<int, IEnumerable<int>, IEnumerable<int>> resultSelector)"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(23394, "https://github.com/dotnet/roslyn/issues/23394")]
+        public async Task QueryMethodinfoJoinInto2()
+        {
+            await TestInMethodAsync(
+@"
+            var q = from i1 in new int[0]
+                    join i2 in new int[0] on i1 equals i2 $$into g
+                    select g;
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(23394, "https://github.com/dotnet/roslyn/issues/23394")]
+        public async Task QueryMethodinfoRangeVariableSimple1()
+        {
+            await TestInMethodAsync(
+@"
+            var q = $$from double i in new int[0]
```

Comment:
Could you add a similar test without the `double` cast, ie. for the vanilla `$$from` scenario?

Reply:
Sure I can. But there is nothing to be shown in that case.

Reply:
Done db7afce.